data-manager/dataframe/src/test/groovy/org/orbisgis/orbisdata/datamanager/dataframe/GroovyDataFrameTest.groovy

```diff
@@ -93,27 +94,45 @@ class GroovyDataFrameTest {
         def h2GIS = RANDOM_DS();
         h2GIS.execute("""
                 DROP TABLE IF EXISTS geotable;
-                CREATE TABLE geotable (id int, the_geom geometry(point), type varchar,temperature int);
-                INSERT INTO geotable VALUES (1, 'POINT(10 10)'::GEOMETRY, 'grass', -12), (2, 'POINT(1 1)'::GEOMETRY, 'corn', 22);
+                CREATE TABLE geotable (id int, the_geom geometry(point), type varchar,temperature int, baby_jeje_weight double);
+                INSERT INTO geotable VALUES (1, 'POINT(10 10)'::GEOMETRY, 'grass', -12, 4.780), (2, 'POINT(1 1)'::GEOMETRY, 'corn', 22, 5.500);
         """)
         DataFrame df = DataFrame.of(h2GIS.select().from("GEOTABLE").where("type = 'grass'").getSpatialTable())
         assertNotNull df
         assertNotNull df.schema()
-        assertEquals 4, df.ncols()
+        assertEquals 5, df.ncols()
         assertEquals 0, df.columnIndex("ID")
         assertEquals 1, df.columnIndex("THE_GEOM")
         assertEquals 2, df.columnIndex("TYPE")
         assertEquals 3, df.columnIndex("TEMPERATURE")
+        assertEquals 4, df.columnIndex("BABY_JEJE_WEIGHT")
         df.save(h2GIS, "output_dataframe", true)
         def table = h2GIS.getTable("OUTPUT_DATAFRAME")
         assertNotNull(table)
-        assertEquals(4,table.getColumnCount())
+        assertEquals(5,table.getColumnCount())
         assertEquals(1,table.getRowCount())
         table.next();
         def row = table.firstRow()
         assertEquals(1,row.ID)
         assertEquals("POINT (10 10)",row.THE_GEOM)
         assertEquals("grass",row.TYPE)
         assertEquals(-12,row.TEMPERATURE)
+        assertEquals(4.780,row.BABY_JEJE_WEIGHT)
+    }
+
+    @Test
+    void testCreateDFFromTable() {
+        def h2GIS = RANDOM_DS();
+        h2GIS.execute("""
+                DROP TABLE IF EXISTS geotable;
+                CREATE TABLE geotable (id int,  type varchar,temperature int, baby_jeje_weight double);
+                INSERT INTO geotable VALUES (1,  'grass', -12, 4.780), (2,  'corn', 22, null);
+        """)
+        DataFrame df = DataFrame.of(h2GIS.select().from("GEOTABLE").where("type = 'corn'").getSpatialTable())
+        assertNotNull df
+        assertEquals(2,df.get(0, 0))
+        assertEquals("corn",df.get(0, 1))
+        assertEquals(22,df.get(0, 2))
+        assertNull(df.get(0, 3))
```

Comment:
This test seems to have some code duplicated with the previous, maybe they could be merged.

Reply:
I prefer to keep this test separate if possible. Because it just tests the creation of the dataframe.